### ISSUE TITLE
Version 1.0.5 does not work on .NET Framework 4.7.2

### ISSUE THREAD

**Issue by IvanILanekassen** [ks-no/fiks-maskinporten-client-dotnet#24]
Hi,

When I try to run v1.0.5 on .NET Framework 4.7.2, the code fails with the following exception:
```
Unhandled Exception: System.Security.Cryptography.CryptographicException: Invalid algorithm specified.

   at System.Security.Cryptography.CryptographicException.ThrowCryptographicException(Int32 hr)
   at System.Security.Cryptography.Utils.SignValue(SafeKeyHandle hKey, Int32 keyNumber, Int32 calgKey, Int32 calgHash, Byte[] hash, Int32 cbHash, ObjectHandleOnStack retSignature)
   at System.Security.Cryptography.Utils.SignValue(SafeKeyHandle hKey, Int32 keyNumber, Int32 calgKey, Int32 calgHash, Byte[] hash)
   at System.Security.Cryptography.RSACryptoServiceProvider.SignHash(Byte[] rgbHash, Int32 calgHash)
   at System.Security.Cryptography.RSACryptoServiceProvider.SignHash(Byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding)
   at System.Security.Cryptography.RSA.SignData(Byte[] data, Int32 offset, Int32 count, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding)
   at System.Security.Cryptography.RSA.SignData(Byte[] data, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding)
   at JWT.Algorithms.RS256Algorithm.Sign(Byte[] bytesToSign) in D:\a\1\s\src\JWT\Algorithms\RS256Algorithm.cs:line 70
   at JWT.Algorithms.RS256Algorithm.Sign(Byte[] key, Byte[] bytesToSign) in D:\a\1\s\src\JWT\Algorithms\RS256Algorithm.cs:line 55
   at JWT.JwtEncoder.Encode(IDictionary`2 extraHeaders, Object payload, Byte[] key) in D:\a\1\s\src\JWT\JwtEncoder.cs:line 59
   at Ks.Fiks.Maskinporten.Client.Jwt.JwtRequestTokenGenerator.CreateEncodedJwt(String scope, MaskinportenClientConfiguration configuration)
   at Ks.Fiks.Maskinporten.Client.MaskinportenClient.CreateRequestContent(TokenRequest tokenRequest)
   at Ks.Fiks.Maskinporten.Client.MaskinportenClient.<GetNewAccessToken>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Ks.Fiks.Maskinporten.Client.MaskinportenClient.<>c__DisplayClass12_0.<<GetAccessTokenForRequest>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Ks.Fiks.Maskinporten.Client.Cache.TokenCache.<UpdateOrAddToken>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Ks.Fiks.Maskinporten.Client.Cache.TokenCache.<GetToken>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Ks.Fiks.Maskinporten.Client.MaskinportenClient.<GetAccessTokenForRequest>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Ks.Fiks.Maskinporten.Client.MaskinportenClient.<GetAccessToken>d__9.MoveNext()
```

This is because of an issue similar to this issue: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1324. It seems to be an issue that was introduced in .NET 4.7. 

To fix the bug, this code has to be used when instantiating the `RS256Algorithm`object in the `JwtRequestTokenGenerator` constructor: `new RS256Algorithm(_certificate.GetRSAPublicKey(), _certificate.GetRSAPrivateKey())`. (The old code is: `new RS256Algorithm(_certificate)`). I can create a PR if you want, but it should be easy to fix. 

The new code fixes the problem because the implementaton of  `GetRSAPrivateKey()` uses some of the same methods that was introduced as an workaround in the AzureAD issue (reference source can be found here: https://github.com/microsoft/referencesource/blob/master/System.Core/System/Security/Cryptography/X509Certificates/RSACertificateExtensions.cs)

Best regards,
Ivan Uthus

**Comment by IvanILanekassen** [comment]
Please take a look at the potential fix in https://github.com/ks-no/fiks-maskinporten-client-dotnet/pull/25.

Best regards,
Ivan Uthus

**Comment by zapodot** [comment]
Confirmed fixed by applying fragment from #25
